### ISSUE TITLE
"separated" builder

### ISSUE THREAD

**Issue by Maatteogekko** [wwwdata/implicitly_animated_reorderable_list#25]
I was looking for a way to use `SliverImplicitlyAnimatedList` like a `SliverList.separated`. This is not possible with the current API, but looking at the code it should be easy to allow it.

I was thinking to allow passing a `delegateBuilder` to `CustomSliverAnimatedList` and use it [here](https://github.com/wwwdata/implicitly_animated_reorderable_list/blob/17cf0ba4a4672a871972fe02181076f87354b038/lib/src/custom_sliver_animated_list.dart#L151C3-L153C4).
Then, for my case at least, I would be able to pass a delegate like this:

```dart
class SliverChildSeparatedBuilderDelegate extends SliverChildBuilderDelegate {
  SliverChildSeparatedBuilderDelegate({
    required NullableIndexedWidgetBuilder itemBuilder,
    ChildIndexGetter? findChildIndexCallback,
    required NullableIndexedWidgetBuilder separatorBuilder,
    int? itemCount,
    bool addAutomaticKeepAlives = true,
    bool addRepaintBoundaries = true,
    bool addSemanticIndexes = true,
  }) : super(
          (context, index) {
            final itemIndex = index ~/ 2;
            final Widget? widget;
            if (index.isEven) {
              widget = itemBuilder(context, itemIndex);
            } else {
              widget = separatorBuilder(context, itemIndex);
              // ignore: prefer_asserts_with_message , we use FlutterError
              assert(() {
                if (widget == null) {
                  throw FlutterError('separatorBuilder cannot return null.');
                }

                return true;
              }());
            }

            return widget;
          },
          findChildIndexCallback: findChildIndexCallback,
          childCount: itemCount == null ? null : math.max(0, itemCount * 2 - 1),
          addAutomaticKeepAlives: addAutomaticKeepAlives,
          addRepaintBoundaries: addRepaintBoundaries,
          addSemanticIndexes: addSemanticIndexes,
          semanticIndexCallback: (_, index) {
            return index.isEven ? index ~/ 2 : null;
          },
        );
}
```

Then we could also easily have `SliverImplicitlyAnimatedList.separated()` and `ImplicitlyAnimatedList.separated()`.

I don't see why this should not be possible, but also haven't tested it yet. I could make a PR for this.

**Comment by wwwdata** [comment]
sounds good. Go for it and open a PR :) Any contribution is welcome 👍

**Comment by wwwdata** [comment]
Thanks a lot again. It's merged in and I released 0.5.1 which has the changes.
